### ISSUE TITLE
TeamObjectiveResource crashing on map load

### ISSUE THREAD

**Issue by Nopey** [openfortress/Open-Fortress-Source-Base#10]
Currently on master branch, loading into arena_badlands (and presumably any map with control points) crashes the game.
This is caused by [`g_pObjectiveResource`](https://github.com/openfortress/Open-Fortress-Source-Base/blob/fb9c5ea181ef3d640461c6467539ec5e7d579306/src/game/server/team_objectiveresource.h#L233) being NULL.

We'll have to create `COFTeamObjectiveResource` (and presumably `C_OFTeamObjectiveResource`).
Reverse engineering the TF one may help, but it's chalk full of MVM garbo.

Stacktrace: 
```cpp
CTeamControlPoint::GetTeamCapPercentage (this=this@entry=0x22469800, iTeam=iTeam@entry=2) at /of/src/src/game/server/team_control_point.cpp:889
889		int iCappingTeam = ObjectiveResource()->GetCappingTeam( GetPointIndex() );
(gdb) bt
#0  CTeamControlPoint::GetTeamCapPercentage (this=this@entry=0x22469800, iTeam=iTeam@entry=2) at /of/src/src/game/server/team_control_point.cpp:889
#1  0xe23b1abc in CTeamControlPoint::UpdateCapPercentage (this=this@entry=0x22469800) at /of/src/src/game/server/team_control_point.cpp:875
#2  0xe23b1d4b in CTeamControlPoint::InternalSetOwner (this=this@entry=0x22469800, iCapTeam=0, bMakeSound=bMakeSound@entry=false, iNumCappers=iNumCappers@entry=0,
    pCappingPlayers=pCappingPlayers@entry=0x0) at /of/src/src/game/server/team_control_point.cpp:620
#3  0xe23b2d47 in CTeamControlPoint::Spawn (this=0x22469800) at /of/src/src/game/server/team_control_point.cpp:134
#4  0xe2406a88 in DispatchSpawn (pEntity=pEntity@entry=0x22469800) at /of/src/src/game/server/util.cpp:1925
#5  0xe22cb78e in SpawnAllEntities (nEntities=189, pSpawnList=0x20e8c000, bActivateEntities=false) at /of/src/src/game/server/mapentities.cpp:270
#6  0xe22cbddf in SpawnHierarchicalList (nEntities=nEntities@entry=189, pSpawnList=pSpawnList@entry=0x20e8c000, bActivateEntities=bActivateEntities@entry=false)
    at /of/src/src/game/server/mapentities.cpp:510
#7  0xe22cc48b in MapEntity_ParseAllEntities (pMapData=<optimized out>,
    pMapData@entry=0x1fe24000 "{\n\"world_maxs\" \"4096 4096 1008\"\n\"world_mins\" \"-4096 -4096 -256\"\n\"skyname\" \"sky_badlands_01\"\n\"maxpropscreenwidth\" \"-1\"\n\"detailvbsp\" \"detail.vbsp\"\n\"detailmaterial\" \"detail/detailsprites\"\n\"classname\" \"wo"..., pFilter=pFilter@entry=0xffffc5ec,
    bActivateEntities=bActivateEntities@entry=false) at /of/src/src/game/server/mapentities.cpp:484
#8  0xe22a4115 in CServerGameDLL::LevelInit (this=0xe2709ec8 <g_ServerGameDLL>, pMapName=0xffffc9b8 "arena_badlands",
    pMapEntities=0x1fe24000 "{\n\"world_maxs\" \"4096 4096 1008\"\n\"world_mins\" \"-4096 -4096 -256\"\n\"skyname\" \"sky_badlands_01\"\n\"maxpropscreenwidth\" \"-1\"\n\"detailvbsp\" \"detail.vbsp\"\n\"detailmaterial\" \"detail/detailsprites\"\n\"classname\" \"wo"..., pOldLevel=0x0, pLandmarkName=0x0, loadGame=false, background=false)
    at /of/src/src/game/server/gameinterface.cpp:1043
```

```
(gdb) print g_pObjectiveResource
$1 = (CBaseTeamObjectiveResource *) 0x0
(gdb) print g_pObjectiveResource->m_iNumControlPoints
Cannot access memory at address 0x358
```

**Comment by Nopey** [comment]
relevant snippet of datamaps.txt
```
CTFObjectiveResource - tf_objective_resource
- m_nMannVsMachineMaxWaveCount (Offset 4016) (Save)(4 Bytes)
- m_nMannVsMachineWaveCount (Offset 4020) (Save)(4 Bytes)
- m_nMannVsMachineWaveEnemyCount (Offset 4024) (Save)(4 Bytes)
- m_nMvMWorldMoney (Offset 4028) (Save)(4 Bytes)
- m_flMannVsMachineNextWaveTime (Offset 4032) (Save)(4 Bytes)
- m_bMannVsMachineBetweenWaves (Offset 4036) (Save)(1 Bytes)
- m_nMannVsMachineWaveClassCounts (Offset 4040) (Save)(48 Bytes)
- m_iszMannVsMachineWaveClassNames (Offset 4136) (Save)(48 Bytes)
- m_nMannVsMachineWaveClassFlags (Offset 4232) (Save)(48 Bytes)
- m_nMannVsMachineWaveClassCounts2 (Offset 4088) (Save)(48 Bytes)
- m_iszMannVsMachineWaveClassNames2 (Offset 4184) (Save)(48 Bytes)
- m_nMannVsMachineWaveClassFlags2 (Offset 4280) (Save)(48 Bytes)
- m_bMannVsMachineWaveClassActive (Offset 4328) (Save)(12 Bytes)
- m_bMannVsMachineWaveClassActive2 (Offset 4340) (Save)(12 Bytes)
- m_iTimerToShowInHUD (Offset 876) (Save)(4 Bytes)
- m_iStopWatchTimer (Offset 880) (Save)(4 Bytes)
- m_iNumControlPoints (Offset 884) (Save)(4 Bytes)
- m_bPlayingMiniRounds (Offset 888) (Save)(1 Bytes)
- m_bControlPointsReset (Offset 889) (Save)(1 Bytes)
- m_iUpdateCapHudParity (Offset 892) (Save)(4 Bytes)
- m_flCustomPositionX (Offset 3744) (Save)(4 Bytes)
- m_flCustomPositionY (Offset 3748) (Save)(4 Bytes)
- m_vCPPositions (Offset 896) (Save)(96 Bytes)
- m_bCPIsVisible (Offset 992) (Save)(32 Bytes)
- m_flLazyCapPerc (Offset 1024) (Save)(32 Bytes)
- m_iTeamIcons (Offset 1056) (Save)(256 Bytes)
- m_iTeamOverlays (Offset 1312) (Save)(256 Bytes)
- m_iTeamReqCappers (Offset 1568) (Save)(256 Bytes)
- m_flTeamCapTime (Offset 1824) (Save)(256 Bytes)
- m_iPreviousPoints (Offset 2080) (Save)(768 Bytes)
- m_bTeamCanCap (Offset 2848) (Save)(64 Bytes)
- m_iTeamBaseIcons (Offset 2912) (Save)(128 Bytes)
- m_iBaseControlPoints (Offset 3040) (Save)(128 Bytes)
- m_bInMiniRound (Offset 3168) (Save)(8 Bytes)
- m_iWarnOnCap (Offset 3176) (Save)(32 Bytes)
- m_iszWarnSound (Offset 3208) (Save)(32 Bytes)
- m_iNumTeamMembers (Offset 3344) (Save)(256 Bytes)
- m_iCappingTeam (Offset 3600) (Save)(32 Bytes)
- m_iTeamInZone (Offset 3632) (Save)(32 Bytes)
- m_bBlocked (Offset 3664) (Save)(8 Bytes)
- m_iOwner (Offset 3672) (Save)(32 Bytes)
- m_bCPCapRateScalesWithPlayers (Offset 3704) (Save)(8 Bytes)
- m_pszCapLayoutInHUD (Offset 3712) (Save)(32 Bytes)
- m_flCapPercentages (Offset 3784) (Save)(32 Bytes)
- m_iCPGroup (Offset 3752) (Save)(32 Bytes)
- m_bCPLocked (Offset 3272) (Save)(8 Bytes)
- m_nNumNodeHillData (Offset 3816) (Save)(16 Bytes)
- m_flNodeHillData (Offset 3832) (Save)(160 Bytes)
- m_bTrackAlarm (Offset 3992) (Save)(4 Bytes)
- m_flUnlockTimes (Offset 3280) (Save)(32 Bytes)
- m_flCPTimerTimes (Offset 3312) (Save)(32 Bytes)
```

**Comment by Nopey** [comment]
and snippet of netprops.txt, excluding those in base class
```
 Member: m_nMannVsMachineMaxWaveCount (offset 4016) (type integer) (bits 9) (Unsigned)
 Member: m_nMannVsMachineWaveCount (offset 4020) (type integer) (bits 9) (Unsigned)
 Member: m_nMannVsMachineWaveEnemyCount (offset 4024) (type integer) (bits 16) (Unsigned)
 Member: m_nMvMWorldMoney (offset 4028) (type integer) (bits 16) (Unsigned)
 Member: m_flMannVsMachineNextWaveTime (offset 4032) (type float) (bits 0) (NoScale)
 Member: m_bMannVsMachineBetweenWaves (offset 4036) (type integer) (bits 1) (Unsigned)
 Member: m_nFlagCarrierUpgradeLevel (offset 4352) (type integer) (bits 4) (Unsigned)
 Member: m_flMvMBaseBombUpgradeTime (offset 4356) (type float) (bits 0) (NoScale)
 Member: m_flMvMNextBombUpgradeTime (offset 4360) (type float) (bits 0) (NoScale)
 Member: m_iszMvMPopfileName (offset 4368) (type string) (bits 0) ()
 Member: m_iChallengeIndex (offset 4364) (type integer) (bits 16) ()
 Member: m_nMvMEventPopfileType (offset 4372) (type integer) (bits 4) (Unsigned)
 Table: m_nMannVsMachineWaveClassCounts (offset 4040) (type m_nMannVsMachineWaveClassCounts)
  Member: 000 (offset 0) (type integer) (bits 16) ()
  ...
  Member: 011 (offset 44) (type integer) (bits 16) ()
 Member: m_iszMannVsMachineWaveClassNames (offset 4136) (type string) (bits 0) (InsideArray)
 Member: m_iszMannVsMachineWaveClassNames (offset 0) (type array) (bits 0) ()
 Table: m_nMannVsMachineWaveClassFlags (offset 4232) (type m_nMannVsMachineWaveClassFlags)
  Member: 000 (offset 0) (type integer) (bits 10) (Unsigned)
  ...
  Member: 011 (offset 44) (type integer) (bits 10) (Unsigned)
 Table: m_nMannVsMachineWaveClassCounts2 (offset 4088) (type m_nMannVsMachineWaveClassCounts2)
  Member: 000 (offset 0) (type integer) (bits 16) ()
  ...
  Member: 011 (offset 44) (type integer) (bits 16) ()
 Member: m_iszMannVsMachineWaveClassNames2 (offset 4184) (type string) (bits 0) (InsideArray)
 Member: m_iszMannVsMachineWaveClassNames2 (offset 0) (type array) (bits 0) ()
 Table: m_nMannVsMachineWaveClassFlags2 (offset 4280) (type m_nMannVsMachineWaveClassFlags2)
  Member: 000 (offset 0) (type integer) (bits 10) (Unsigned)
  ...
  Member: 011 (offset 44) (type integer) (bits 10) (Unsigned)
 Table: m_bMannVsMachineWaveClassActive (offset 4328) (type m_bMannVsMachineWaveClassActive)
  Member: 000 (offset 0) (type integer) (bits 1) (Unsigned)
  ...
  Member: 011 (offset 11) (type integer) (bits 1) (Unsigned)
 Table: m_bMannVsMachineWaveClassActive2 (offset 4340) (type m_bMannVsMachineWaveClassActive2)
  Member: 000 (offset 0) (type integer) (bits 1) (Unsigned)
  ...
  Member: 011 (offset 11) (type integer) (bits 1) (Unsigned)
```
(Posting these to demonstrate how the tf2 class is almost entirely mvm.

**Comment by CherrySodaPop** [comment]
Implemented COFObjectiveResource in [b7a8ceb](https://github.com/openfortress/Open-Fortress-Source-Base/commit/b7a8cebc47621330486919de2fdac407f70f7b97)
